.github/workflows/test-and-push.yml

```diff
@@ -39,6 +39,8 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
+        with:
+          ref: 'refs/heads/master'
```

Comment:
When re-running an old action, we could potentially be running the tests on different version of the code than the one being pushed to the server  
I feel like we should keep the `test` job consistent with `push` and make it checkout the `master` branch when the action is being run on that branch

Reply:
I think it would be super weird to run tests on a different commit than the one that triggered the build. 

I added an extra check on the job to compare the head of the target branch with the commit that triggered the build and only run publishing code if the refs match.  
Example "test" build: https://github.com/dianabarsan/gha-action-on-master/actions/workflows/compare-sha.yml